src/lib.rs

```diff
@@ -333,29 +335,45 @@ impl<T: Ord> OrderedCollection<T> {
         X: Ord,
     {
         let x = x.borrow();
-        let mut i = 0;
+        let mut i = 1;
+
         let mask = prefetch_mask(self.items.len());
+        // the search loop is proven to be CPU-backend and not memory bound when using prefetch. So offset part
```

Comment:
I'd avoid the word "proven" here. Maybe:
```suggestion
        // the search loop is arithmetic-bound, not memory-bound when using prefetch. So offset part
```

Reply:
You are definitely right! I chose wrong wording here. This is an observation which is dependent on particular microarchitecture.